iXBRLViewerPlugin/viewer/src/js/search.js

```diff
@@ -107,12 +144,17 @@ export class ReportSearch {
         var results = []
         var searchIndex = this;
 
+        const filters = [
+            this.visibilityFilter,
+            this.periodFilter,
+            this.conceptTypeFilter,
+            this.factValueFilter,
+            this.calculationsFilter
+        ];
+
         rr.forEach((r,i) => {
                 var item = searchIndex._report.getItemById(r.ref);
-                if ((item.isHidden() ? s.showHiddenFacts : s.showVisibleFacts) &&
-                    (s.periodFilter === '*' || item.period().key() === s.periodFilter) &&
-                    (s.conceptTypeFilter === '*' || s.conceptTypeFilter === (item.isNumeric() ? 'numeric' : 'text')) &&
-                    (s.factValueFilter === '*' || (s.factValueFilter === 'positive' && item.isPositive()) || (s.factValueFilter === 'negative' && item.isNegative())))
+                if (filters.every(f => f(s, item)))
                 {
```

Comment:
nit: pre-existing, suggest moving this `{` to the preceding line as it was before `factValueFilter`.